service/gcs/main.go

```diff
@@ -5,21 +5,60 @@ import (
 	"fmt"
 	"io"
 	"os"
+	"path/filepath"
 	"syscall"
 	"time"
 
+	"github.com/Microsoft/opengcs/internal/cgrouputils"
 	"github.com/Microsoft/opengcs/internal/oc"
 	"github.com/Microsoft/opengcs/internal/runtime/hcsv2"
 	"github.com/Microsoft/opengcs/service/gcs/bridge"
 	"github.com/Microsoft/opengcs/service/gcs/core/gcs"
 	"github.com/Microsoft/opengcs/service/gcs/runtime/runc"
 	"github.com/Microsoft/opengcs/service/gcs/transport"
 	"github.com/containerd/cgroups"
+	v1 "github.com/containerd/cgroups/stats/v1"
 	oci "github.com/opencontainers/runtime-spec/specs-go"
 	"github.com/sirupsen/logrus"
 	"go.opencensus.io/trace"
 )
 
+func memoryLogFormat(metrics *v1.Metrics) logrus.Fields {
+	return logrus.Fields{
+		"memory-usage":       metrics.Memory.Usage.Usage,
+		"memory-usage-max":   metrics.Memory.Usage.Max,
+		"memory-usage-limit": metrics.Memory.Usage.Limit,
+		"swap-usage":         metrics.Memory.Swap.Usage,
+		"swap-usage-max":     metrics.Memory.Swap.Max,
+		"swap-usage-limit":   metrics.Memory.Swap.Limit,
+		"kernel-usage":       metrics.Memory.Kernel.Usage,
+		"kernel-usage-max":   metrics.Memory.Kernel.Max,
+		"kernel-usage-limit": metrics.Memory.Kernel.Limit,
+	}
+}
+
+func readMemoryEvents(efdFile *os.File, cgName string, threshold int64, cg cgroups.Cgroup) {
+	// Buffer must be >= 8 bytes for eventfd reads
+	// http://man7.org/linux/man-pages/man2/eventfd.2.html
+	buf := make([]byte, 8)
+	for {
+		_, err := efdFile.Read(buf)
+		if err != nil {
+			logrus.WithError(err).Warn("opengcs::readEvents - failed to read from eventfd")
```

Comment:
I don't think we should keep using this prefix like `opengcs::readEvents - ` on our log statements. It's something I want to start removing since it usually just seems to add noise.

Reply:
We should have a pr to do something similar in vain to https://github.com/microsoft/hcsshim/pull/791 this eventually as well then

Reply:
I'm in favor of continuing to improve log statements as we come across them. We don't necessarily have to do them all at once.